src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/LoadCoinviewRule.cs

```diff
@@ -22,18 +21,14 @@ public class SaveCoinviewRule : UtxoStoreConsensusRule
         /// <inheritdoc />
         public override async Task RunAsync(RuleContext context)
         {
-            uint256 oldBlockHash = context.ValidationContext.ChainedHeaderToValidate.Previous.HashBlock;
-            uint256 nextBlockHash = context.ValidationContext.ChainedHeaderToValidate.HashBlock;
+            ChainedHeader currentBlock = context.ValidationContext.ChainedHeaderToValidate;
 
             // Persist the changes to the coinview. This will likely only be stored in memory,
-            // unless the coinview treashold is reached.
+            // unless the coinview threshold is reached.
             this.Logger.LogTrace("Saving coinview changes.");
-            var utxoRuleContext = context as UtxoRuleContext;
-            await this.PowParent.UtxoSet.SaveChangesAsync(utxoRuleContext.UnspentOutputSet.GetCoins(this.PowParent.UtxoSet), null, oldBlockHash, nextBlockHash).ConfigureAwait(false);
-
-            bool forceFlush = this.FlushRequired(context.ValidationContext.ChainedHeaderToValidate);
-            if (this.PowParent.UtxoSet is CachedCoinView cachedCoinView)
-                await cachedCoinView.FlushAsync(forceFlush).ConfigureAwait(false);
+            var utxoRuleContext = (UtxoRuleContext)context;
+            List<UnspentOutputs> unspentOutputs = utxoRuleContext.UnspentOutputSet.GetCoins(this.PowParent.UtxoSet)?.ToList();
+            await this.PowParent.UtxoSet.AddRewindDataAsync(unspentOutputs, currentBlock).ConfigureAwait(false);
         }
 
         /// <summary>
```

Comment:
So this method is not used anymore?

Reply:
Correct. There is no more manual flushing. It is handled by internal task, same as BSQ